plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,373 @@ var plugin = {},
+    /*
+    * @apiName: GetFeedbackData
+    * @apiDescription: Get feedback data with or without filters
+    * @apiParam: 'widget_id', Id of related widget
+    * @apiParam: 'rating', filter by rating
+    * @apiParam: 'device_id', filter by device_id
+    * @apiParam: 'app_id', app_id of related application
+    */
+    plugins.register('/o/web-feedback/data', function(ob) {
+        var params = ob.params;
+        var app = params.qstring.app_id;
+        var collectionName = 'feedback' + app;
+        var query = {};
+        
+        if (params.qstring.widget_id) {
+            query["widget_id"] = params.qstring.widget_id;
+        }
+        if (params.qstring.rating) {
+            query["rating"] = params.qstring.rating;
+        }
+        if (params.qstring.device_id) {
+            query["device_id"] = params.qstring.device_id;
+        }
+
+        common.db.collection(collectionName)
+            .find(query)
+            .toArray(function(err, docs) {
+                if (!err) {
+                    common.returnOutput(params, docs);
+                    return true;
+                } else {
+                    common.returnMessage(params, 500, err.message);
+                    return false;
+                }
+            })
+
+        return true;
+    });
 
-    plugins.register("/i", ratingEventProcess);
+    /*
+    * @apiName: GetWidgetsData
+    * @apiDescription: Get feedback widgets with or without filters
+    * @apiParam: 'app_id', app_id of related application
+    * @apiParam: 'is_active', is_active option for widgets
+    */
+    plugins.register('/o/web-feedback/widgets',  function(ob) {
+        var params = ob.params;
+        common.db.collection('apps')
+            .findOne({'key': params.qstring.app_key}, (err, app) => {
+                if (!app) {
+                    if (plugins.getConfig("api").safe) {
+                        common.returnMessage(params, 400, 'App does not exist');
+                    }
+                    return false;
+                }
+                
+                var collectionName = 'web_feedback_widgets_' + app["_id"];
+                var query = {};
+                
+                if (params.qstring.is_active) {
+                    query["is_active"] = params.qstring.is_active;
+                }
+
+                common.db.collection(collectionName)
+                    .find(query)
+                    .toArray(function(err, docs) {
+                        if (!err) {
+                            common.returnOutput(params, docs);
+                            return true;
+                        } else {
+                            common.returnMessage(params, 500, err.message);
+                            return false;
+                        }
+                    })
+            })
+        return true;
+    });
+
+    /*
+    * @apiName: GetOneWidget
+    */
+    plugins.register('/o/web-feedback/widget', function(ob) {
```

Comment:
I assume this one is for fetching form for SDK, so we use app_key and no validation, correct?
So it means only person, who has widget_id can fetch the data.